xilem/src/view/flex.rs

```diff
@@ -139,61 +138,263 @@ where
+        let (pod, state) = self.view.build(cx);
+        (
+            FlexElement::Child(Box::new(pod.inner.boxed().into()), self.params),
+            state,
+        )
+    }
+
+    fn rebuild<'el>(
+        &self,
+        prev: &Self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        mut element: Mut<'el, Self::Element>,
+    ) -> Mut<'el, Self::Element> {
+        {
+            if self.params != prev.params {
+                element
+                    .parent
+                    .update_child_flex_params(element.idx, self.params);
+            }
+            let mut child = element
+                .parent
+                .child_mut(element.idx)
+                .expect("FlexWrapper always has a widget child");
+            self.view
+                .rebuild(&prev.view, view_state, ctx, child.downcast());
+        }
+        element
+    }
+
+    fn teardown(
+        &self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        mut element: Mut<'_, Self::Element>,
+    ) {
+        let mut child = element
+            .parent
+            .child_mut(element.idx)
+            .expect("FlexWrapper always has a widget child");
+        self.view.teardown(view_state, ctx, child.downcast());
+    }
+
+    fn message(
+        &self,
+        view_state: &mut Self::ViewState,
+        id_path: &[xilem_core::ViewId],
+        message: xilem_core::DynMessage,
+        app_state: &mut State,
+    ) -> xilem_core::MessageResult<Action> {
+        self.view.message(view_state, id_path, message, app_state)
+    }
+}
+
+#[derive(Copy, Clone, PartialEq)]
+pub enum FlexSpacer {
+    Fixed(f64),
+    Flex(f64),
+}
+
+impl<State, Action> View<State, Action, ViewCtx> for FlexSpacer {
+    type Element = FlexElement;
+
+    type ViewState = ();
+
+    fn build(&self, _ctx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
+        let el = match self {
+            FlexSpacer::Fixed(len) => FlexElement::FixedSpacer(*len),
+            FlexSpacer::Flex(flex) => FlexElement::FlexSpacer(*flex),
+        };
+        (el, ())
+    }
+
+    fn rebuild<'el>(
+        &self,
+        prev: &Self,
+        _: &mut Self::ViewState,
+        _: &mut ViewCtx,
+        mut element: Mut<'el, Self::Element>,
+    ) -> Mut<'el, Self::Element> {
+        if self != prev {
+            match self {
+                FlexSpacer::Fixed(len) => element.parent.update_spacer_fixed(element.idx, *len),
+                FlexSpacer::Flex(flex) => element.parent.update_spacer_flex(element.idx, *flex),
+            };
+        }
+        element
+    }
+
+    fn teardown(&self, _: &mut Self::ViewState, _: &mut ViewCtx, _: Mut<'_, Self::Element>) {}
+
+    fn message(
+        &self,
+        _: &mut Self::ViewState,
+        _: &[xilem_core::ViewId],
+        _: DynMessage,
+        _: &mut State,
+    ) -> xilem_core::MessageResult<Action> {
+        unreachable!()
     }
 }
```

Comment:
I think we're missing `AnyFlexView` here.

We could also support a `FlexChild` view, which would be:
```rust
enum FlexChild<V>
{
    Child(V, ...),
    Spacer(...)
}
```
although I don't think that has much value if we have `AnyFlexView`.

Reply:
You mean, that one could do something like this?:

```rust
fn my_flex_elem() -> FlexChild {...}
```

which would be dynamic, so that it could be for example used in a `Vec`?

Reply:
Well, either:
```rust
fn my_flex_elem() -> FlexChild<Box<AnyWidgetView<MyState>>> {...}
```
or
```rust
fn my_flex_elem() -> AnyFlexView<MyState> {...}
```
But essentially that, yes.